nand2tetris.el

```diff
@@ -51,76 +51,82 @@
 
 
 ;;; Scripts Integration
-(defcustom nand2tetris-tools-dir
-  (expand-file-name "tools" nand2tetris-core-base-dir)
+(defcustom nand2tetris-tools-dir nil
```

Comment:
For each of these variables I used `ag` to search for all references and changed them to call the getter function instead, so I think I've got all of them.

Reply:
This `getter function` is new to me. What's the difference over just having a custom variable called `nand2tetris-tools-dir` ?

What I can think of is that if one changes `nand2tetris-core-base-dir` then the variables would "update" automatically. 

Actually, that seems enough reason. Do you think it's still important to have a variable `nand2tetris-tools-dir` then? It seems redundant now that I think of it in terms of this getter function.

Reply:
Yeah it's so that you can change the value of the base dir after loading nand2tetris, which is generally much nicer.

I left the variables there because I don't want to break anyone's existing setup. I know I get annoyed when I update a package and suddenly something doesn't work!